### PR TITLE
[*] CORE : Enable debug mode from BO

### PR DESCRIPTION
**Pull request by firstred**
## Description

Enable debug mode from the Back Office. 
Especially useful for those folks out there who are requesting for remote assistance, be it on the forum or somewhere else.
## Steps to Test this Fix

- Set correct permissions on the file `/config/defines.inc.php` and check if `PS_MODE_DEV` has been defined
- Check the error messages with incorrect write permissions, wrong read permissions, missing define, etc.

### CODE REVIEW

**Comment by PrestaEdit** [comment]
I didn't read all the changes code but, one question: why not use the "defines_custom.inc.php" file here ? :)

**Comment by firstred** [comment]
Would that be better?

**Comment by PrestaEdit** [comment]
I think yes. At least, handle the two files (if the first exist and the define are in, it's the first to be handle by PrestaShop). As I read, If you define the mode dev in the custom file, this code can't handle it ;-)

**Comment by firstred** [comment]
Thanks for the info! Let me see if I can update the PR.

**Comment by firstred** [comment]
Btw, does it have to be named `defines_custom.inc.php` ? Aren't all files ending with `.inc.php` processed?

**Comment by firstred** [comment]
Nope, found the answer in `/config/config.inc.php`. It has to be named `defines_custom.inc.php`.

**Comment by firstred** [comment]
Alright, fixed. If the file `defines_custom.inc.php` defines `_PS_MODE_DEV_` then this file will be used, otherwise `defines.inc.php`.

**Comment by vincentbz** [comment]
Tested and ok, thanks !

But as it could be dangerous for merchants, we will wait for a specific content from our content manager with warnings, before merging it :)

**Comment by firstred** [comment]
Hmmz, in that case you might also want to add a warning to the "Disable overrides" option, I suppose.

**Comment by mickaelandrieu** [comment]
@AlexEven there is a "waiting for wording" label, do we need you before merge this contribution ?

**Comment by AlexEven** [comment]
@mickaelandrieu Yes, that's the purpose of the label ;)
You can merge, sorry for the delay, it's all good. Thanks!

**Comment by mickaelandrieu** [comment]
@firstred do you need to add something on this contrib ? For me it's ready for production :+1:

**Comment by firstred** [comment]
Nope :)

File: controllers/admin/AdminPerformanceController.php [hunk 1]
```diff
@@ -1008,6 +1037,33 @@ public function postProcess()
...
+            if (Tools::isSubmit('debug_mode') && (bool)Tools::getValue('debug_mode')) {
+                $debug_mode_status = $this->enableDebugMode();
+            } else {
+                $debug_mode_status = $this->disableDebugMode();
+            }
+
+            if (!empty($debug_mode_status)) {
+                switch ($debug_mode_status) {
+                    case self::DEBUG_MODE_ERROR_COULD_NOT_BACKUP:
+                        $this->errors[] = Tools::displayError(sprintf($this->l('Error: could not write to file: %s. Make sure that the file or directory is writable.'), _PS_ROOT_DIR_.'/config/defines.old.php'));
```
Comment (PrestaEdit) [inline controllers/admin/AdminPerformanceController.php]: defines.old.php doesn't exist, DEBUG_MODE_ERROR_COULD_NOT_BACKUP never used.

**Comment by mickaelandrieu** [comment]
@firstred thank you ;)

File: controllers/admin/AdminPerformanceController.php (hunk 1 above)
Comment (PrestaEdit) [inline controllers/admin/AdminPerformanceController.php]: DEBUG_MODE_ERROR_COULD_NOT_BACKUP not set, too.

**Comment by mickaelandrieu** [comment]
@PrestaEdit noooooo :'( can you make a little patch ?

**Pull request merged**